020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/RelativePathNonAppEndpointTest.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.qe.non_application.endpoint;
+
+import java.util.Collections;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.qe.http.non_application.endpoint.HelloResource;
+import io.quarkus.test.QuarkusProdModeTest;
+
+public class RelativePathNonAppEndpointTest extends CommonNonAppEndpoint {
+    private static final String BASE_PATH = "q";
+
+    @RegisterExtension
+    static final QuarkusProdModeTest relativePathScenario = new QuarkusProdModeTest()
+            .overrideConfigKey("quarkus.http.root-path", "/api")
+            .overrideConfigKey("quarkus.http.non-application-root-path", BASE_PATH)
+            .overrideConfigKey("quarkus.http.redirect-to-non-application-root-path", "false")
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(HelloResource.class));
+
+    @BeforeEach
+    public void beforeEach() {
+        relativePathScenario.stop();
+        relativePathScenario.start();
```

Comment:
This should not be necessary if we set `.setRun(true)` in `QuarkusProdModeTest`, however I see that for native this is not working properly. I will investigate it further and report it if needed. 
No need to change anything.